app/src/main/java/com/example/triptracker/model/repository/ImageRepository.kt

```diff
@@ -32,4 +33,23 @@ class ImageRepository {
       Response.Failure(e)
     }
   }
+
+  suspend fun addProfilePictureToFirebaseStorage(imageUri: Uri): Response<Uri> {
+    return try {
+      val downloadUrl =
+          storage.reference
+              .child(PICTURE_FOLDER)
+              .child(PROFILE_PICTURES)
+              .child(UUID.randomUUID().toString())
+              .putFile(imageUri)
+              .await()
+              .storage
+              .downloadUrl
+              .await()
+      Log.d("DOWNLOAD URL", downloadUrl.toString())
+      Response.Success(downloadUrl)
+    } catch (e: Exception) {
+      Response.Failure(e)
+    }
+  }
```

Comment:
Maybe we need to modularize this function in some future sprints so that we avoid code duplication. For now it's fine !